internal/show/versions.go

```diff
@@ -67,19 +94,119 @@ func printDeploymentVersions(clusterInfo *cluster.Info, printer *table.Printer,
 		}
 
 		version, repository := images.ParseOperatorImage(deployment.Spec.Template.Spec.Containers[0].Image)
-		printer.Add(component, repository, version)
+
+		runningVersion, err := getVersionForComponent(clusterInfo, component,
+			labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels).String())
+		if err != nil {
+			return err
+		}
+
+		printer.Add(component, repository, version, runningVersion)
 	}
 
 	return nil
 }
 
+func getVersionForComponent(clusterInfo *cluster.Info, component, labelSelector string) (string, error) {
+	podsClient := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Pods(constants.OperatorNamespace)
+	podList, err := podsClient.List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+
+	if err != nil || len(podList.Items) < 1 {
+		return "", errors.Wrapf(err, "Failed to find pods for component %s", component)
+	}
+
+	// Try all pods
+	for i := range podList.Items {
+		pod := podList.Items[i]
+		podVersion, _ := getVersionFromPodBinary(&pod, clusterInfo, component)
+
+		if podVersion != "" {
+			return podVersion, nil
+		}
+
+		podVersion = getVersionFromPodLogs(&pod, podsClient, component)
+
+		if podVersion == "" {
+			continue
+		}
+
+		return podVersion, nil
+	}
+
+	return "", errors.Errorf("Failed to get version for %s", component)
+}
+
+func getVersionFromPodBinary(pod *corev1.Pod, clusterInfo *cluster.Info, component string) (string, error) {
+	cmdArg := "--version"
+
+	switch component {
+	case names.LighthouseCoreDNSComponent:
+		cmdArg = "--subm-version"
+	case names.MetricsProxyComponent:
+		cmdArg = "version"
+	}
```

Comment:
This means we end up encoding information about the commands to use in two places: the map at the start of the file, and here. I think it would be better to specify everything in the map (make it a map of strings to string slices).